set-balance.md

```diff
@@ -168,6 +192,346 @@ A `Result` is considered an `Action`.
+        ...
+        </k>
+```
+
+Helpers for calling `set_free_balance` and `set_reserved_balance`.
+
+* Sets the new free balance
+* Emits an imbalance event
+* Helper function for `set_balance`
+
+```k
+    syntax Action ::= "set_balance_free"     "(" AccountId "," Int ")"
+    syntax Action ::= "set_balance_reserved" "(" AccountId "," Int ")"
+ // ------------------------------------------------------------------
+    rule [balance-set-free]:
+         <k> set_balance_free(WHO, FREE_BALANCE') => set_free_balance(WHO, FREE_BALANCE') ... </k>
+         <totalIssuance> ISSUANCE => ISSUANCE +Int (FREE_BALANCE' -Int FREE_BALANCE) </totalIssuance>
+         <account>
+           <accountID> WHO </accountID>
+           <freeBalance> FREE_BALANCE </freeBalance>
+           ...
+         </account>
+      requires #inWidth(64, ISSUANCE +Int (FREE_BALANCE' -Int FREE_BALANCE))
+
+    rule [balance-set-reserved]:
+         <k> set_balance_reserved(WHO, RESERVED_BALANCE') => set_reserved_balance(WHO, RESERVED_BALANCE') ... </k>
+         <totalIssuance> ISSUANCE => ISSUANCE +Int (RESERVED_BALANCE' -Int RESERVED_BALANCE) </totalIssuance>
+         <account>
+           <accountID> WHO </accountID>
+           <reservedBalance> RESERVED_BALANCE </reservedBalance>
+           ...
+         </account>
+      requires #inWidth(64, ISSUANCE +Int (RESERVED_BALANCE' -Int RESERVED_BALANCE))
+```
+
+### `transfer`
+
+Transfer some liquid free balance to another account.
+
+`transfer` will set the `FreeBalance` of the sender and receiver.
+It will decrease the total issuance of the system by the `TransferFee`.
+If the sender's account is below the existential deposit as a result
+of the transfer, the account will be reaped.
+
+The dispatch origin for this call must be `Signed` by the transactor.
+
+```k
+    syntax ExitenceRequirement ::= ".AllowDeath"
+                                 | ".KeepAlive"
+
+    syntax Action ::= transfer(AccountId, AccountId, Int)
+ // ---------------------------------------------------------------------
+    rule [transfer-self]:
+         <k> transfer(ORIGIN, ORIGIN, _) => . ... </k>
+
+    rule [transfer-existing-account]:
+         <k> transfer(ORIGIN, DESTINATION, AMOUNT)
+          => set_free_balance(ORIGIN, SOURCE_BALANCE -Int AMOUNT -Int FEE)
+          ~> set_free_balance(DESTINATION, DESTINATION_BALANCE +Int AMOUNT)
+         ...
+         </k>
+         <totalIssuance> ISSUANCE => ISSUANCE -Int FEE </totalIssuance>
+         <transferFee> FEE </transferFee>
+         <accounts>
+           <account>
+             <accountID> ORIGIN </accountID>
+             <freeBalance> SOURCE_BALANCE </freeBalance>
+             ...
+           </account>
+           <account>
+             <accountID> DESTINATION </accountID>
+             <freeBalance> DESTINATION_BALANCE </freeBalance>
+             ...
+           </account>
+         </accounts>
+      requires ORIGIN =/=K DESTINATION
+       andBool DESTINATION_BALANCE >Int 0
```

Comment:
This side condition seems weird to me. Is it to make sure it's not a dust or empty account?

Reply:
Correct.  The fee for creating an account is different from that for sending funds to an existing account.